### PR TITLE
feat: unexpected request type check for exception middleware

### PR DESCRIPTION
**Pull request by thorseraq**

### CODE REVIEW

File: apps/server/src/middleware/exception-logger.ts [hunk 1]
```diff
@@ -16,6 +16,13 @@ export class ExceptionLogger implements ExceptionFilter {
   catch(exception: Error, host: ArgumentsHost) {
     // with useGlobalFilters, the context is always HTTP
     const ctx = host.switchToHttp();
+
+    // TODO: temporary, remove this after stable
+    if (host.getType() !== 'http') {
+      this.logger.error(`Unexpected request type caught: ${host.getType()}`);
```
Comment (Brooooooklyn) [inline apps/server/src/middleware/exception-logger.ts:22]: provide more information here, maybe request path or something that helpful to debug

**Review by darkskygit — APPROVED** [review]

**Review by Brooooooklyn — COMMENTED** [review]
I have a question, this ExceptionLogger is used to log exceptions, but why does it seem to change the original response, for example I see a lot of response.json calls, why is it handled this way instead of returning the response as is?

**Comment by thorseraq** [comment]
That's the best way I think, but I also tried to return response directly, the http request will not get response (If exception was caught by it)

**Comment by thorseraq** [comment]
throw an exception -> app crash
return an exception -> crash
return response -> no response
reponse.json() -> works

File: apps/server/src/middleware/exception-logger.ts [hunk 2]
```diff
@@ -4,31 +4,60 @@ import {
...
+    const ifVerboseLog = !TrivialExceptions.some(e => exception instanceof e);
     this.logger.error(
       new Error(
         `${requestId ? `requestId-${requestId}:` : ''}${exception.message}`,
         { cause: exception }
-      ),
-      exception.stack
+      )
     );
+    ifVerboseLog && this.logger.error(exception.stack);
```
Comment (Brooooooklyn) [inline apps/server/src/middleware/exception-logger.ts:49]:
```suggestion
    if (ifVerboseLog) {
      this.logger.error(exception.stack);
    }
```

**Comment by thorseraq** [comment]
closed by https://github.com/toeverything/AFFiNE/pull/4209

**Pull request closed without merging**